draft-normington-jsonpath.md

```diff
@@ -166,14 +166,22 @@ The terminology of {{-json}} applies.
 
 Data Item:
 : A structure complying to the generic data model of JSON, i.e.,
-  composed of containers such as arrays and maps (JSON objects), and
-  of atomic data such as null, true, false, numbers, and text strings.
+  composed of containers, namely maps (JSON objects) and arrays, and
+  of atomic data, namely null, true, false, numbers, and text strings.
+  Also called a JSON value.
 
 Object:
 : Used in its generic sense, e.g., for programming language objects.
   When a JSON Object as defined in {{-json}} is meant, we specifically
   say JSON Object.
 
+Member:
+: A key/value pair in a map (JSON Object).  (Not itself a JSON value.)
```

Comment:
Perhaps we should define "key" and even declare "property" as a synonym.

Reply:
Property is a synonym for member, as is entry and field.
How many of these synonyms do we need?
(RFC 8259 calls the keys "names", which is not better, except maybe in combinations such as "field name" or "member name".)

Reply:
"key" and "property" are by far the most common terms I see in use.

Reply:
And I've been caught calling keys "labels".  Some people like to call them "tags".
My first hit when searching was https://www.w3schools.com/Js/js_json_objects.asp, which you may like or not like (they do confuse JSON objects with their representation), but for them introducing JSON objects as key/value pairs seemed obvious, not needing further explanation.

Reply:
Key/value is indeed very common. "field" is less common but nevertheless attested in the wild.

I tend to prefer `field name/value` mainly for two reasons: a) it builds on top of RFC 8259 terminology b) the "field" qualifier makes sense also when referring to the value (the _field value_)

Reply:
In the spirit of "you can't please everyone", my own opinion from an implementer's viewpoint is that use of synonyms is best avoided. Implementer's want a parsimonious vocabulary with precise definitions, preferably one that's consistent with the core specifications (RFC 8259), the introduction of synonyms which are "suggestive" only frightens them. Words mean different things in different contexts, for example, "field" in the Java specifications refers to a type and an identifier, in a CODASYL data record it means something else. "property" isn't mentioned in RFC 8259, why introduce it here?      

Regarding "perhaps we should define 'key'", if it appears in the ABNF, it is defined.

Reply:
It probably would help to keep "name" around for names in jsonpath.
But given that "key" is not defined in 8259, I'll stick with "name" for now, expanding to "member name" when needed.